homeassistant/components/github/sensor.py

```diff
@@ -1,293 +1,355 @@
-        return self.issues_response.data[0] if self.issues_response.data else None
+    def native_value(self) -> StateType:
+        """Return the state of the sensor."""
+        return self.coordinator.data.issues[0].title[:255]
 
     @property
-    def last_pull_request(self):
-        """Return the last pull request."""
-        return self.pulls_response.data[0] if self.pulls_response.data else None
+    def extra_state_attributes(self) -> Mapping[str, str | int | None]:
+        """Return the extra state attributes."""
+        issue = self.coordinator.data.issues[0]
+        return {
+            "url": issue.html_url,
+            "number": issue.number,
+        }
+
+
+class GitHubSensorLatestPullEntity(GitHubSensorLatestBaseEntity):
+    """Defines a GitHub latest pull sensor entity."""
+
+    _coordinator_key: CoordinatorKeyType = "issue"
+    _name: str = "Latest Pull Request"
+
+    coordinator: RepositoryIssueDataUpdateCoordinator
 
     @property
-    def last_release(self):
-        """Return the last release."""
-        return self.releases_response.data[0] if self.releases_response.data else None
-
-    async def async_update(self):
-        """Update GitHub data."""
-        try:
-            await asyncio.gather(
-                self._update_repository(),
-                self._update_commit(),
-                self._update_issues(),
-                self._update_pulls(),
-                self._update_releases(),
-            )
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        return super().available and len(self.coordinator.data.pulls) != 0
 
-            if self.repository_response.data.permissions.push:
-                await asyncio.gather(
-                    self._update_clones(),
-                    self._update_views(),
-                )
-
-            self.available = True
-        except GitHubException as err:
-            _LOGGER.error("GitHub error for %s: %s", self.repository_path, err)
-            self.available = False
-
-    async def _update_repository(self):
-        """Update repository data."""
-        self.repository_response = await self._github.repos.get(self.repository_path)
-
-    async def _update_commit(self):
-        """Update commit data."""
-        self.commit_response = await self._github.repos.list_commits(
-            self.repository_path, **{"params": {"per_page": 1}}
-        )
+    @property
+    def native_value(self) -> StateType:
+        """Return the state of the sensor."""
+        return self.coordinator.data.pulls[0].title[:255]
 
-    async def _update_issues(self):
-        """Update issues data."""
-        self.issues_response = await self._github.repos.issues.list(
-            self.repository_path
-        )
+    @property
+    def extra_state_attributes(self) -> Mapping[str, str | int | None]:
+        """Return the extra state attributes."""
+        pull = self.coordinator.data.pulls[0]
```

Comment:
Can we adjust the coordinators to only fetch the latest pull instead of all ?

Reply:
We need all to determine the count.